soroban-env-common/src/env.rs

```diff
@@ -168,6 +168,10 @@ macro_rules! call_macro_with_all_host_functions {
                 /// vector contains the contract id as Hash, and a function as
                 /// a Symbol.
                 {"8", fn get_current_call_stack() -> Object }
+                /// Causes the currently executing contract to fail immediately
+                /// with a provided status code, which must be of error-type
+                /// `ScStatusType::ContractError`. Does not actually return.
+                {"9", fn fail_with_status(status:Status) -> RawVal }
```

Comment:
I'm wondering if fail is the best term for us to use. Maybe we should use `abort_with_`, since what will happen is equivalent to aborting? This isn't particularly important and it is fine, and maybe preferable, as is.